File: flutter_app/lib/countries.dart
/// MDH davlatlari — telefon raqami kiritishdan oldin davlat tanlash uchun
/// (davlat kodi + bayroq emoji + nom). Backend `PhoneOTP`/`User.phone`
/// har qanday formatdagi qatorni qabul qiladi, shuning uchun bu yerda faqat
/// UI darajasida raqamga kod qo'shib beriladi va uzunligi tekshiriladi
/// (davlat kodidan keyingi mahalliy raqam necha xonali bo'lishi kerak).
class CountryInfo {
  final String name;
  final String dialCode;
  final String flag;
  final int phoneLength;
  const CountryInfo(this.name, this.dialCode, this.flag, this.phoneLength);

  bool isValid(String localNumber) =>
      RegExp(r'^\d+$').hasMatch(localNumber) &&
      localNumber.length == phoneLength;
}

const List<CountryInfo> cisCountries = [
  CountryInfo("O'zbekiston", '+998', '🇺🇿', 9),
  CountryInfo('Rossiya', '+7', '🇷🇺', 10),
  CountryInfo("Qozog'iston", '+7', '🇰🇿', 10),
  CountryInfo("Qirg'iziston", '+996', '🇰🇬', 9),
  CountryInfo('Tojikiston', '+992', '🇹🇯', 9),
  CountryInfo('Turkmaniston', '+993', '🇹🇲', 8),
  CountryInfo('Ozarbayjon', '+994', '🇦🇿', 9),
  CountryInfo('Armaniston', '+374', '🇦🇲', 8),
  CountryInfo('Belarus', '+375', '🇧🇾', 9),
  CountryInfo('Moldova', '+373', '🇲🇩', 8),
];

File: ios/FurniturePlatform/Sources/Networking/Countries.swift
import Foundation

/// MDH davlatlari — telefon raqami kiritishdan oldin davlat tanlash uchun.
/// Flutter'dagi `lib/countries.dart` bilan bir xil.
struct CountryInfo: Identifiable, Equatable {
    let name: String
    let dialCode: String
    let flag: String
    let phoneLength: Int
    var id: String { name }

    func isValid(_ localNumber: String) -> Bool {
        localNumber.count == phoneLength && localNumber.allSatisfy(\.isNumber)
    }
}

let cisCountries: [CountryInfo] = [
    CountryInfo(name: "O'zbekiston", dialCode: "+998", flag: "🇺🇿", phoneLength: 9),
    CountryInfo(name: "Rossiya", dialCode: "+7", flag: "🇷🇺", phoneLength: 10),
    CountryInfo(name: "Qozog'iston", dialCode: "+7", flag: "🇰🇿", phoneLength: 10),
    CountryInfo(name: "Qirg'iziston", dialCode: "+996", flag: "🇰🇬", phoneLength: 9),
    CountryInfo(name: "Tojikiston", dialCode: "+992", flag: "🇹🇯", phoneLength: 9),
    CountryInfo(name: "Turkmaniston", dialCode: "+993", flag: "🇹🇲", phoneLength: 8),
    CountryInfo(name: "Ozarbayjon", dialCode: "+994", flag: "🇦🇿", phoneLength: 9),
    CountryInfo(name: "Armaniston", dialCode: "+374", flag: "🇦🇲", phoneLength: 8),
    CountryInfo(name: "Belarus", dialCode: "+375", flag: "🇧🇾", phoneLength: 9),
    CountryInfo(name: "Moldova", dialCode: "+373", flag: "🇲🇩", phoneLength: 8),
]
